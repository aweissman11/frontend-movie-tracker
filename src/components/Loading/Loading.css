.triangle {
  position: relative;
  width: 0;
  height: 0;
  border-top: 40px solid transparent;
  border-bottom: 40px solid transparent;
  border-left: 60px solid #6CC4EE;
  background: transparent;
  animation: spin-triangle 5s infinite;
}
.triangle h3 {
  top: -60px;
  left: -63px;
  position: absolute;
  font-family: 'Open Sans', sans-serif;
  font-size: 35px;
  font-weight: 800;
  color: #fff;
  animation: spin-text 5s infinite;
}

@keyframes spin-text {
  0% {
    transform: rotateY(0)
  }
  20% {
    transform: rotateY(-180deg)
  }
  40% {
    transform: rotateY(-180deg)
  }
  60% {
    transform: rotateY(-360deg)
  }
  80% {
    transform: rotateY(-360deg)
  }
  100% {
    transform: rotateY(-360deg)
  }
}

@keyframes spin-triangle {
  0% {
    transform: rotateY(0)
  }
  20% {
    transform: rotateY(0)
  }
  40% {
    transform: rotateY(180deg)
  }
  60% {
    transform: rotateY(180deg)
  }
  80% {
    transform: rotateY(360deg)
  }
  100% {
    transform: rotateY(360deg)
  }
}