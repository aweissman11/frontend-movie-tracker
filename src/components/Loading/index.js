import React from 'react'

import './Loading.css'

const Loading = () => {
  return(
  <div class='triangle'>
    <h3>MT</h3>
  </div>
  )
}