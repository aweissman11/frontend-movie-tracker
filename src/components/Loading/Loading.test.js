import React from 'react';
import Loading from './index';
import { shallow } from 'enzyme';

describe('Loading', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<Loading />)
    expect(wrapper).toMatchSnapshot()
  })
})