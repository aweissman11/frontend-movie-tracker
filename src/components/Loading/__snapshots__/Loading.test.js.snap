// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Loading />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "aria-label": "loading",
      "children": <h3>
        MT
      </h3>,
      "className": "triangle",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "MT",
      },
      "ref": null,
      "rendered": "MT",
      "type": "h3",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "aria-label": "loading",
        "children": <h3>
          MT
        </h3>,
        "className": "triangle",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "MT",
        },
        "ref": null,
        "rendered": "MT",
        "type": "h3",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
